core/rfb.js

```diff
@@ -1623,6 +1627,31 @@ export default class RFB extends EventTargetMixin {
         return this._fail("No supported sub-auth types!");
     }
 
+    _negotiateRA2neAuth() {
+        if (this._rfbRA2State === null) {
+            this._rfbRA2State = new RA2State(this._sock, () => {
+                this._rfbInitState = "SecurityResult";
+                this._rfbRA2State = null;
+                this._initMsg();
+            }, () => this._rfbCredentials, (hasUsername, fingerprint) => {
+                if (hasUsername) {
+                    this.dispatchEvent(new CustomEvent(
+                        "credentialsrequired",
+                        { detail: { types: ["username", "password"], fingerprint: fingerprint } }));
```

Comment:
In order for the API to remain stable in the future you should also include a type with the server key. So that new types can be added for other things that need to be verified.